src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,170 @@
+package duke;
+
+import duke.enums.Commands;
+import duke.enums.TaskTypes;
+import duke.exceptions.DukeException;
+import duke.parser.Parser;
+import duke.storage.TaskStorage;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.tasklist.Tasklist;
+
+import java.time.LocalDate;
+
+import java.util.ArrayList;
+
+/**
+ * Duke is the personal assistant for managing user
+ * tasks such as todos, deadlines, and events.
+ */
+public class Duke {
+    private static final String FILE_PATH = "data/data.txt";
+    private Parser parser;
+    private final Tasklist tasklist;
+    private boolean isActive;
+
+    public Duke() {
+        this.parser = new Parser();
+        this.tasklist = new Tasklist(new TaskStorage(FILE_PATH));
+    }
+
+    /**
+     * Runs Duke and begins parsing input from the user.
+     * Duke will deactivate and exit upon receiving "bye"
+     * from user input.
+     */
+    public void run() {
+        this.isActive = true;
+    }
+
+    public String getResponse(String input) {
```

Comment:
it's great to see that you have added javadocs to a lot of your methods, maybe when you have time, you could add them for these methods too?